orttraining/orttraining/python/training/onnxblock/_training_graph_utils.py

```diff
@@ -106,10 +116,14 @@ def build_gradient_graph(
     eval_model = copy.deepcopy(model)
     _disable_training_mode(eval_model)
 
-    optimized_model = onnx.load_from_string(get_optimized_model(model.SerializeToString(), requires_grad))
+    options = SessionOptions()
+    if custom_op_library is not None:
```

Comment:
curious why we need set this before build gradient graph.

Reply:
The graph optimization and gradient graph building, both require us to load the ONNX model into ONNX Runtime's `Model` class. However, loading an ONNX model with custom op definitions will fail since ONNX Runtime cannot understand which op the node corresponds to (without having access to the schema).
This is why we need to pass in the custom op library to ONNX Runtime while optimizing the graph and building the gradient graph.